libtiledbsoma/src/reindexer/reindexer.cc

```diff
@@ -90,7 +82,9 @@ void IntIndexer::lookup(const int64_t* keys, int64_t* results, int size) {
     if (size == 0) {
         return;
     }
-    if (tiledb_thread_pool_ == nullptr) {  // When concurrency is 1
+    // Single thread checks
+    if (context_ == nullptr || context_->thread_pool() == nullptr ||
+        context_->thread_pool()->concurrency_level() == 1) {
```

Comment:
I think we can just check `context_->thread_pool()->concurrency_level() == 1`

Reply:
I get your point but in cases like this, I really prefer to be ultra conservative